<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CwmsFunctionEditorClass</class>
 <widget class="QWidget" name="CwmsFunctionEditorClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1068</width>
    <height>669</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Funktionseditor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="JSEdit" name="m_pjsEditor"/>
       </item>
       <item>
        <widget class="QGroupBox" name="m_pqgbSearch">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Suche</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="leftMargin">
           <number>3</number>
          </property>
          <property name="topMargin">
           <number>3</number>
          </property>
          <property name="rightMargin">
           <number>3</number>
          </property>
          <property name="bottomMargin">
           <number>3</number>
          </property>
          <item>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Suchen</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="m_pqleSearch"/>
            </item>
            <item row="0" column="2">
             <widget class="QPushButton" name="m_pqpbNext">
              <property name="toolTip">
               <string>NächsterTreffer</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../resources/emresources.qrc">
                <normaloff>:/icons/arrow-right24.png</normaloff>:/icons/arrow-right24.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QPushButton" name="m_pqpbPRevious">
              <property name="toolTip">
               <string>Vorheriger Treffer</string>
              </property>
              <property name="text">
               <string/>
              </property>
              <property name="icon">
               <iconset resource="../resources/emresources.qrc">
                <normaloff>:/icons/arrow-left24.png</normaloff>:/icons/arrow-left24.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>24</width>
                <height>24</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Ersetzen</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="m_pqleReplace"/>
            </item>
            <item row="1" column="2">
             <widget class="QPushButton" name="m_pqpbReplace">
              <property name="toolTip">
               <string>Ersetzen</string>
              </property>
              <property name="text">
               <string>Ersetzen</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QPushButton" name="m_pqpbReplaceAll">
              <property name="toolTip">
               <string>Alle Ersetzen</string>
              </property>
              <property name="text">
               <string>Alle Ersetzen</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QCheckBox" name="m_pqchbCaseSensitive">
              <property name="text">
               <string>Groß Kleinschreibung berücksichtigen</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="m_pqchbWholeWord">
              <property name="text">
               <string>Nach ganzen Wörtern suchen</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="m_pqchbRegExp">
              <property name="text">
               <string>Regulärer Ausdruck</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>400</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="m_pqwProperties">
       <attribute name="title">
        <string>Eigenschaften</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <property name="leftMargin">
         <number>3</number>
        </property>
        <property name="topMargin">
         <number>3</number>
        </property>
        <property name="rightMargin">
         <number>3</number>
        </property>
        <property name="bottomMargin">
         <number>3</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="m_pqlName">
          <property name="text">
           <string>Name</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLineEdit" name="m_pqleName"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Rückgabewert</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QComboBox" name="m_pqcbReturnType"/>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Aufrufparameter</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <property name="leftMargin">
            <number>3</number>
           </property>
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="rightMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item row="0" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Typ</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="m_pqcbType"/>
             </item>
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Name</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="m_pqleParameterName"/>
             </item>
             <item>
              <widget class="QLabel" name="m_pqlClass">
               <property name="text">
                <string>Klasse</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="m_pqcbClass"/>
             </item>
             <item>
              <widget class="QPushButton" name="m_pqpbAddParameter">
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../resources/emresources.qrc">
                 <normaloff>:/icons/new24.png</normaloff>:/icons/new24.png</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>24</width>
                 <height>24</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <widget class="QTreeWidget" name="m_pqtwParameters">
             <column>
              <property name="text">
               <string>Pos</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Typ</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Name</string>
              </property>
             </column>
            </widget>
           </item>
           <item row="2" column="0">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QPushButton" name="m_pqpbDeleteParameter">
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../resources/emresources.qrc">
                 <normaloff>:/icons/delete24.png</normaloff>:/icons/delete24.png</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>24</width>
                 <height>24</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="m_pqpbMoveParameterUp">
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../resources/emresources.qrc">
                 <normaloff>:/icons/arrow-up24.png</normaloff>:/icons/arrow-up24.png</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>24</width>
                 <height>24</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="m_pqpbMoveParameterDown">
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="../resources/emresources.qrc">
                 <normaloff>:/icons/arrow-down24.png</normaloff>:/icons/arrow-down24.png</iconset>
               </property>
               <property name="iconSize">
                <size>
                 <width>24</width>
                 <height>24</height>
                </size>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="0" colspan="3">
         <widget class="QGroupBox" name="m_pqgbFunctionType">
          <property name="title">
           <string>Funktionsart</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QRadioButton" name="m_pqrbObject">
             <property name="text">
              <string>Memberfunktion</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="m_pqrbStatic">
             <property name="text">
              <string>Statisch</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item row="4" column="0" colspan="3">
         <widget class="QGroupBox" name="m_pqgbAccess">
          <property name="title">
           <string>Zugriffsart</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <property name="leftMargin">
            <number>3</number>
           </property>
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="rightMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item>
            <widget class="QRadioButton" name="m_pqrbPublic">
             <property name="text">
              <string>Öffentlich</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="m_pqrbProtected">
             <property name="text">
              <string>Geschützt</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="m_pqrbPrivate">
             <property name="text">
              <string>Privat</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QPushButton" name="m_pqpbRights">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../resources/emresources.qrc">
            <normaloff>:/icons/usermanager24.png</normaloff>:/icons/usermanager24.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QPushButton" name="m_pqpbSyntaxCheck">
          <property name="toolTip">
           <string>Syntaxprüfung</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../resources/emresources.qrc">
            <normaloff>:/icons/articleunit.png</normaloff>:/icons/articleunit.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>24</width>
            <height>24</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="m_pqwDocumentation">
       <attribute name="title">
        <string>Dokumentation</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <property name="leftMargin">
         <number>3</number>
        </property>
        <property name="topMargin">
         <number>3</number>
        </property>
        <property name="rightMargin">
         <number>3</number>
        </property>
        <property name="bottomMargin">
         <number>3</number>
        </property>
        <item row="0" column="0">
         <widget class="QGroupBox" name="m_pqgbComment">
          <property name="title">
           <string>Dokumentation</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <property name="leftMargin">
            <number>3</number>
           </property>
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="rightMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Darstellungsname</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="m_pqleCaption"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Icon</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="m_pqpbSelectIcon">
             <property name="text">
              <string>...</string>
             </property>
             <property name="iconSize">
              <size>
               <width>24</width>
               <height>24</height>
              </size>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="m_pqgbFunctionDescription">
          <property name="title">
           <string>Funktionsbeschreibung</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <property name="leftMargin">
            <number>3</number>
           </property>
           <property name="topMargin">
            <number>3</number>
           </property>
           <property name="rightMargin">
            <number>3</number>
           </property>
           <property name="bottomMargin">
            <number>3</number>
           </property>
           <item>
            <widget class="QPlainTextEdit" name="m_pqteComment"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>JSEdit</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">jsedit.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/emresources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>m_pqpbSyntaxCheck</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>SyntaxCheckClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>754</x>
     <y>657</y>
    </hint>
    <hint type="destinationlabel">
     <x>372</x>
     <y>779</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbRights</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>RightsClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>716</x>
     <y>657</y>
    </hint>
    <hint type="destinationlabel">
     <x>597</x>
     <y>564</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqcbType</sender>
   <signal>currentIndexChanged</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>TypeIndexChangedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>746</x>
     <y>125</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>64</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbDeleteParameter</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>DeleteParameterClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>707</x>
     <y>514</y>
    </hint>
    <hint type="destinationlabel">
     <x>597</x>
     <y>219</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbAddParameter</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>AddParameterClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1049</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>597</x>
     <y>15</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbMoveParameterDown</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>MoveParameterDownClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>783</x>
     <y>514</y>
    </hint>
    <hint type="destinationlabel">
     <x>597</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbSelectIcon</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>SelectIconClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>840</x>
     <y>89</y>
    </hint>
    <hint type="destinationlabel">
     <x>1249</x>
     <y>132</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqleReplace</sender>
   <signal>returnPressed()</signal>
   <receiver>m_pqpbReplace</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>628</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>625</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqleSearch</sender>
   <signal>returnPressed()</signal>
   <receiver>m_pqpbNext</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>282</x>
     <y>599</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>602</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbNext</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>FindNextClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>566</x>
     <y>602</y>
    </hint>
    <hint type="destinationlabel">
     <x>564</x>
     <y>717</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbPRevious</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>FindPreviousClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>650</x>
     <y>602</y>
    </hint>
    <hint type="destinationlabel">
     <x>707</x>
     <y>720</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbReplace</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>ReplaceClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>566</x>
     <y>625</y>
    </hint>
    <hint type="destinationlabel">
     <x>583</x>
     <y>750</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbReplaceAll</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>ReplaceAllClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>650</x>
     <y>625</y>
    </hint>
    <hint type="destinationlabel">
     <x>605</x>
     <y>705</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_pqpbMoveParameterUp</sender>
   <signal>clicked()</signal>
   <receiver>CwmsFunctionEditorClass</receiver>
   <slot>MoveParameterUpClickedSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>732</x>
     <y>504</y>
    </hint>
    <hint type="destinationlabel">
     <x>811</x>
     <y>727</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>AddParameterClickedSlot()</slot>
  <slot>DeleteParameterClickedSlot()</slot>
  <slot>MoveParameterUpClickedSlot()</slot>
  <slot>MoveParameterDownClickedSlot()</slot>
  <slot>EnlargeCodeViewClickedSlot()</slot>
  <slot>RightsClickedSlot()</slot>
  <slot>CreateNewFunctionClickedSlot()</slot>
  <slot>SyntaxCheckClickedSlot()</slot>
  <slot>TypeIndexChangedSlot()</slot>
  <slot>SelectIconClickedSlot()</slot>
  <slot>FindNextClickedSlot()</slot>
  <slot>FindPreviousClickedSlot()</slot>
  <slot>ReplaceClickedSlot()</slot>
  <slot>ReplaceAllClickedSlot()</slot>
 </slots>
</ui>
